xpart/particles/particles.py

```diff
@@ -698,53 +952,16 @@ def update_delta(self, new_delta_value):
 
         """
         Update the `delta` value of the particles object. `ptau` and `rvv` and
-        `rpp` are updated accordingly.
+        `rpp` are updated accordingly. If `new_delta_value` contains nans, these
+        values are not updated.
         """
 
-        ctx = self._buffer.context
-
-        if (self._contains_lost_or_unallocated_particles()
-                or _contains_nan(new_delta_value, ctx)):
-            if isinstance(self._buffer.context, xo.ContextPyopencl):
-                # Needs special treatment because masking does not work with pyopencl
-                # could be done with a kernel in the future
-                state_cpu = self.state.get()
-                if hasattr(new_delta_value, 'get'):
-                    new_delta_value_cpu = new_delta_value.get()
-                else:
-                    new_delta_value_cpu = new_delta_value
-                mask_cpu = ((state_cpu > 0) & (~np.isnan(new_delta_value_cpu)))
-                mask = ctx.nparray_to_context_array(np.where(mask_cpu)[0])
-            else:
-                mask = ((self.state > 0) & (~ctx.nplike_lib.isnan(new_delta_value)))
-        else:
-            mask = None
+        mask = (new_delta_value == new_delta_value) & (self.state > 0)
```

Comment:
this is a trick to get isnan on pyopencl (make an explicit function in nplike_lib or at least add a comment